content/hardware/03.nano/boards/nano-matter/tutorials/08.ml-magic-wand/content.md

```diff
@@ -0,0 +1,663 @@
+          Serial.println("*   *   *");
+          Serial.println(" * * * * ");
+          Serial.println("  *   *  ");
+
+          // Green leds for W gesture
+          for (int i = 0; i < 10; i++) { 
+            for (int j = 0; j < 8; j++) {
+              setPixel(j, GREEN);
+            }
+            delay(200); 
+
+            for (int j = 0; j < 8; j++) {
+              setPixel(j, OFF);
+            }
+            delay(200); 
+          }
+          break;
+        case RING_GESTURE:
+          Serial.println("detection = ring (O)");
+          Serial.println("  *****  ");
+          Serial.println(" *     * ");
+          Serial.println("*       *");
+          Serial.println(" *     * ");
+          Serial.println("  *****  ");
+
+          // Yellow leds for O gesture
+          for (int i = 0; i < 10; i++) { 
+            for (int j = 0; j < 8; j++) {
+              setPixel(j, YELLOW);
+            }
+            delay(200); 
+
+            for (int j = 0; j < 8; j++) {
+              setPixel(j, OFF);
+            }
+            delay(200); 
+          }
+          break;
+        case NO_GESTURE:
+          Serial.println("No gesture");
+          break;
+      }
+    }
+
+    // reset LEDs to steady state (blue)
+    for (int i = 0; i < 8; i++) {
+      setPixel(i, BLUE);
+    }
+  } else {
+    printf("error: inference failed");
+  }
+}
+``` 
+
+In addition, some helper functions are used:
+
+- `accelerometer_setup()`: This function initialize the Modulino Movement accelerometer and returns true if it did it successfully.
+- `accelerometer_read(acc_data_t* dst, int n)`: This function read the accelerometer data, normalize it and prepare the model input buffer.
+
+### Upload the Project Sketch
+
+You can download the code from [here](assets/magic_wand_modulino.zip) or by clicking on the image below: 
+
+[![ ](assets/download.png)](assets/magic_wand_modulino.zip)
+
+In the Arduino IDE select the **Arduino Nano Matter** inside the _Silicon Labs_ board package and make sure the **Protocol Stack** is set to _None_.
+
+![Nano Matter Sketch Upload](assets/code.png)
+
+After the code is uploaded successfully, you can test your Nano Matter Magic Wand.
+
+## Project Testing
+
+Power the Nano Matter of your Magic Wand using a USB cable and start moving it following the **Wing** or **Ring** gesture to see if it recognizes it.
+
+<div style="text-align: center;">
+  <video width="100%" controls="true" autoplay loop>
+  <source src="assets/animation2.mp4" type="video/mp4"/>
+  </video>
+</div>
+
+## Conclusion
+
+In this tutorial you learned how to use the Arduino Nano Matter as a Machine Learning gesture recognizer, leveraging the optimized Silicon Labs library for Tiny Machine Learning we showcased the board capabilities of running TinyML algorithms with ease thanks to the **hardware acceleration** achieved with the **Mathematical Vector Processor (MVP)** integrated in the Nano Matter microcontroller.
```

Comment:
_In this tutorial, you learned how to use the Arduino Nano Matter as a Machine Learning gesture recognizer; leveraging the optimized Silicon Labs library for Tiny Machine Learning, we showcased the board capabilities of running TinyML algorithms with ease thanks to the hardware acceleration achieved with the Mathematical Vector Processor (MVP) integrated into the Nano Matter microcontroller._